lib/tether/tether.go

```diff
@@ -173,128 +184,237 @@ func (t *tether) cleanup() {
 	t.ops.Cleanup()
 }
 
-func (t *tether) Start() error {
-	defer trace.End(trace.Begin("main tether loop"))
+func (t *tether) setLogLevel() {
+	// TODO: move all of this into an extension.Pre() block when we move to that model
+	// adjust the logging level appropriately
+	switch t.config.DebugLevel {
+	case 0:
+		log.SetLevel(log.InfoLevel)
+		// TODO: do not echo application output to console without debug enabled
+		serial.DisableTracing()
+	case 1:
+		log.SetLevel(log.DebugLevel)
+		serial.EnableTracing()
```

Comment:
Maybe no tracing on 1?
Also would be good to call:
trace.DisableTracing() or trace.EnableTracing() accordingly.